libs/sdk-common/src/input_parser.rs

```diff
@@ -197,6 +218,47 @@ pub async fn parse(
     Err(anyhow!("Unrecognized input type"))
 }
 
+async fn bip353_parse(
+    input: &str,
+    dns_resolver: &AsyncResolver<GenericConnector<TokioRuntimeProvider>>,
+) -> Option<String> {
+    if let Some((local_part, domain)) = input.split_once('@') {
+        let dns_name = format!("{}.{}.{}", local_part, USER_BITCOIN_PAYMENT_PREFIX, domain);
+
+        // Query for TXT records of a domain
+        let txt_data = match dns_resolver.txt_lookup(dns_name).await {
+            Ok(records) => records
+                .iter()
+                .flat_map(|record| record.to_string().into_bytes())
+                .collect::<Vec<u8>>(),
+            Err(e) => {
+                eprintln!("Failed to lookup TXT records: {}", e);
+                return None;
+            }
+        };
+
+        // Decode TXT data
+        match String::from_utf8(txt_data) {
+            Ok(decoded) => {
+                // The idea is to return from this parse only the BIP21 address and do logic on it inside the parse_core function.
+                // return Some(decoded);
+                if let Some((_, bolt12_address)) = decoded.split_once(BOLT12_PREFIX) {
+                    return Some(bolt12_address.to_string());
+                }
+
+                if let Some((_, lnurl)) = decoded.split_once(LNURL_PAY_PREFIX) {
+                    return Some(lnurl.to_string());
+                }
```

Comment:
The spec states that any value of the txt record should start with "bitcoin:" prefix so we should validate that before attempting to extract the relevant part.
If we are dealing here with either ln offer or lnurl then IMO we should validate it starts with either:
bitcoin:?lno=
bitcoin:?lnurl=

Reply:
@roeierez It's a BIP21 string, which could technically include both. I would say use the existing logic for BIP21 parsing here (with the `querystring::querify` part). Then check if the querystring contains bolt12. If it does: return bolt12. Then check whether it contains lnurl. If it does, return that.

Reply:
That makese sense, I didn't realize it is a bip21 format...
So the validation should be just parse using bip21::Uri and then extract the relevant part as @JssDWt suggested. I agree.

Reply:
As I commented bellow it seems that in the spec the example is missing the bitcoin address (only contains the query params which should serve as fallback).
I suspect the bip21::Uri will fail to parse this string. Perhaps we should attempt to parse it using the crate and if fails try to extract the query params in case it starts with "bitcoin:?" ?